crates/red_knot_python_semantic/src/types/infer.rs

```diff
@@ -1722,6 +1725,22 @@ impl<'db> TypeInferenceBuilder<'db> {
                     _ => Type::Unknown, // TODO
                 }
             }
+            Type::BytesLiteral(lhs) => {
```

Comment:
It occurs to me now that it might be nicer to flatten this out by matching on a `(lhs, rhs, op)` tuple instead, but that would change the existing `IntLiteral` code and is out of scope for this PR